uber/automated_emails.py

```diff
@@ -113,19 +320,35 @@ def __init__(self, conf):
         AutomatedEmail.__init__(self, Attendee,
                                 subject='{EVENT_NAME} Department Checklist: ' + conf.name,
                                 template='shifts/dept_checklist.txt',
-                                filter=lambda a: a.is_single_dept_head and a.admin_account and days_before(7, conf.deadline) and not conf.completed(a),
+                                filter=lambda a: a.is_single_dept_head and a.admin_account and not conf.completed(a),
+                                when=days_before(7, conf.deadline),
                                 sender=c.STAFF_EMAIL,
                                 extra_data={'conf': conf})
 
-before = lambda dt: bool(dt) and localized_now() < dt
-after = lambda dt: bool(dt) and localized_now() > dt
-days_after = lambda days, dt: bool(dt) and (localized_now() > dt + timedelta(days=days))
 
+"""
+IMPORTANT NOTES FOR CHANGING/ADDING EMAIL CATEGORIES:
+
+'ident' is a unique ID for that email category that must not change after emails in that category have started to send.
+If an 'ident' is not set (not recommended), 'ident' will default to be the 'subject' line.
+
+************************************************************************
+IF IDENT IS NOT SET ON A CATEGORY, AND IF YOU CHANGE THE SUBJECT,
+IT WILL CAUSE ANY EMAILS THAT HAVE ALREADY SENT TO RE-SEND.
+************************************************************************
+
+WORK WE NEED TO DO: Going forward, every single email category below should have an explicit 'ident' set. However,
+mid-year we can't change the idents effectively because it will cause emails in these categories to be re-sent.
+
+ident naming RULES:
+ - every single email below should have an explicit ident paramater set. some don't and we should fix that up.
+ - if no 'ident' is set, the ident is set to the 'subject'
+ - name the ident anything you want, ideally something related to the email category i.e. "reg confirmation email"
+ - HOWEVER, IF an ident was not explicitly sent, and you want to rename the subject, you must:
+   - set the ident to the OLD subject first (do not include any {EVENT_NAME} in the ident)
```

Comment:
This would be slightly more clear if it said:
- set the ident to exactly what the old subject is, without using variable names (e.g., "MAGFest 2017" instead of "{EVENT_NAME}")